### PR TITLE
fix: Postrgres group aggregation

### PR DESCRIPTION
**Pull request by srameshr**

### CODE REVIEW

**Comment by srameshr** [comment]
@dplewis @cbaker6 
Here is a new PR

**Comment by dplewis** [comment]
@srameshr Can you add tests?

**Comment by srameshr** [comment]
@dplewis There are already tests for group by, limit, sort, where. My changes do not affect the existing tests. Also all our local code works and returns the right values. So, I assume we do not need tests for this one.

File: src/Adapters/Storage/Postgres/PostgresStorageAdapter.js [hunk 1]
```diff
@@ -2370,7 +2370,18 @@ export class PostgresStorageAdapter implements StorageAdapter {
...
+      columns.forEach((e, i, a) => {
+        if (e && e.trim() === '*') {
+          a[i] = '';
+        }
+      });
+    }
+
+    const qs = `SELECT ${columns
+      .filter(Boolean)
+      .join()} FROM $1:name ${wherePattern} ${skipPattern} ${groupPattern} ${sortPattern} ${limitPattern}`;
```
Comment (dplewis) [inline src/Adapters/Storage/Postgres/PostgresStorageAdapter.js]: Can you explain the problem you are trying to fix? Just so we can determine if there are side-effects.

Comment (srameshr) [reply src/Adapters/Storage/Postgres/PostgresStorageAdapter.js]: @dplewis 
The queries used to fail when * was used with GROUPBY operators and the order of few operators where off when GROUPBY was present.

Comment (srameshr) [reply src/Adapters/Storage/Postgres/PostgresStorageAdapter.js]: @dplewis 
Can you please merge this? All the existing test cases that was there to handle groupBy and other conditions are working as it sits.

Comment (srameshr) [reply src/Adapters/Storage/Postgres/PostgresStorageAdapter.js:2383]: @dplewis Anything on this?

**Comment by dplewis** [comment]
@srameshr Can you resolve the conflict so that we can get this in?

**Comment by cbaker6** [comment]
> @srameshr Sorry for the late reply. Can you resolve the conflict so that we can get this into the next release?

My guess is it's conflicting with the code I added in #6506 because we were editing the same method/lines (note, I'm not able to see the conflict, so I'm guessing). 

@srameshr if this is the case, then the conflict is somewhere around this [line](https://github.com/parse-community/parse-server/blob/dcba8e0b7ca55e55286c21e6ed7c417b0f483103/src/Adapters/Storage/Postgres/PostgresStorageAdapter.js#L2392):

You code: 

``` 
if (groupPattern) {
      columns.forEach((e, i, a) => {
        if (e && e.trim() === '*') {
          a[i] = '';
        }
      });
    }
```

Should be added before the line I mentioned above and that should resolve the conflict (if that's really the conflict) with everything still working. Let me know if that helps.

**Comment by srameshr** [comment]
@cbaker6 Are your changes in master?

**Comment by srameshr** [comment]
@dplewis @cbaker6 
I have fixed the conflicts and updated the code. Can you please look into this? We have a release o n Monday and waiting on it.

**Comment by cbaker6** [comment]
> @dplewis @cbaker6
> I have fixed the conflicts and updated the code. Can you please look into this? We have a release o n Monday and waiting on it.

I think you forgot to lint

**Comment by srameshr** [comment]
@cbaker6 I did link. Its a pre commit hook that automatically runs

**Comment by srameshr** [comment]
@cbaker6 
I did run npm run lint again it is just messing up this line
https://github.com/parse-community/parse-server/pull/6522/files#diff-b0af2ee2c26859a8aa6d8d0a0cf3a64aR2587

**Comment by cbaker6** [comment]
> @cbaker6
> I did run npm run lint again it is just messing up this line
> https://github.com/parse-community/parse-server/pull/6522/files#diff-b0af2ee2c26859a8aa6d8d0a0cf3a64aR2587

Are you doing `npm run lint` locally before you push your code? It's easier to handle the problem locally before pushing. Whenever my lint passes locally, I don't have an issue on CI

**Comment by srameshr** [comment]
@cbaker6
Yes, running it locally and it passes fine. But when I run commit, the line I have mentioned throws a lint error every time. Regardless of how I change it, so I just did a --no-verify while commit and left it to be merged.

File: src/Adapters/Storage/Postgres/PostgresStorageAdapter.js [hunk 2]
```diff
@@ -2575,10 +2584,10 @@ export class PostgresStorageAdapter implements StorageAdapter {
...
           // Cast the error into the proper parse error
           throw new Parse.Error(
-            Parse.Error.DUPLICATE_VALUE,
-            'A duplicate value for a field with unique values was provided'
-          );
-        } else {
+			Parse.Error.DUPLICATE_VALUE,
+			'A duplicate value for a field with unique values was provided'
+		  );
+		} else {
```
Comment (dplewis) [inline src/Adapters/Storage/Postgres/PostgresStorageAdapter.js:2590]: Use 2 spaces instead of tabs

Comment (srameshr) [reply src/Adapters/Storage/Postgres/PostgresStorageAdapter.js:2590]: It's the same thing. I tried setting tab indentation to two spaces and it still fails
```

✖ eslint found some errors. Please fix them and try committing again.

/Users/siddharthramesh/Desktop/code/open-source/parse-server/src/Adapters/Storage/Postgres/PostgresStorageAdapter.js
  2587:1  error  Expected indentation of 8 spaces but found 10  indent
  2588:1  error  Expected indentation of 6 spaces but found 8   indent

✖ 2 problems (2 errors, 0 warnings)
  2 errors and 0 warnings potentially fixable with the `--fix` option.
```

Can you merge it and again re-commit with the right formatting?

Comment (srameshr) [reply src/Adapters/Storage/Postgres/PostgresStorageAdapter.js:2590]: @dplewis @cbaker6 
```npm run lint --fix``` return no error:
```

But running git commit without no-verify fails

Comment (dplewis) [reply src/Adapters/Storage/Postgres/PostgresStorageAdapter.js:2590]: I ran into the same issue you did. Its weird, I know of a fix but its not recommended.

**Comment by srameshr** [comment]
@dplewis @acinader Can I expect a new release this week with the above changes going in?

**Comment by srameshr** [comment]
@dplewis Any timeline as to when the next release will be? Could you do it this week?

**Comment by srameshr** [comment]
@dplewis @acinader Are you guys planning a release this week? Is there a way we can create a release? I am waiting for this because our AWS instances do not allow npm install to run from github repo links for some reason.

**Pull request merged**